# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/xorg-server/restore-z-xaa-nv-compat.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- xorg-server-21.1.21/hw/xfree86/common/xf86Privstr.h.vanilla
+++ xorg-server-21.1.21/hw/xfree86/common/xf86Privstr.h
@@ -78,8 +78,6 @@
     Bool miscModInDevEnabled;   /* Allow input devices to be
                                  * changed */
     Bool miscModInDevAllowNonLocal;
-    Pix24Flags pixmap24;
-    MessageType pix24From;
     Bool pmFlag;
     MessageType iglxFrom;
     XF86_GlxVisuals glxVisuals;
@@ -99,6 +97,8 @@
     Bool autoAddGPU;
     const char *debug;
     Bool autoBindGPU;
+    Pix24Flags pixmap24;
+    MessageType pix24From;
 } xf86InfoRec, *xf86InfoPtr;
 
 /* ISC's cc can't handle ~ of UL constants, so explicitly type cast them. */
--- xorg-server-21.1.21/hw/xfree86/common/xf86Bus.h.vanilla
+++ xorg-server-21.1.21/hw/xfree86/common/xf86Bus.h
@@ -51,10 +51,10 @@
     Bool active;
     Bool inUse;
     BusRec bus;
-    int lastScrnFlag;
     DevUnion *entityPrivates;
     int numInstances;
     GDevPtr *devices;
+    int lastScrnFlag;
 } EntityRec, *EntityPtr;
 
 #define ACCEL_IS_SHARABLE 0x100
--- xorg-server-21.1.21/hw/xfree86/common/xf86str.h.vanilla
+++ xorg-server-21.1.21/hw/xfree86/common/xf86str.h
@@ -594,7 +594,6 @@
     PixmapFormatRec fbFormat;
 
     int bitsPerPixel;           /* fb bpp */
-    Pix24Flags pixmap24;        /* pixmap pref for depth 24 */
     int depth;                  /* depth of default visual */
     MessageType depthFrom;      /* set from config? */
     MessageType bitsPerPixelFrom;       /* set from config? */
@@ -699,6 +698,7 @@
     int reservedInt[NUM_RESERVED_INTS];
     void *reservedPtr[NUM_RESERVED_POINTERS];
     funcPointer reservedFuncs[NUM_RESERVED_FUNCS];
+    Pix24Flags pixmap24;        /* pixmap pref for depth 24 */
 } ScrnInfoRec;
 
 typedef struct {
--- xorg-server-21.1.21/fb/fb.h.vanilla
+++ xorg-server-21.1.21/fb/fb.h
@@ -433,14 +433,14 @@
 
 /* private field of a screen */
 typedef struct {
-    unsigned char win32bpp;     /* window bpp for 32-bpp images */
-    unsigned char pix32bpp;     /* pixmap bpp for 32-bpp images */
 #ifdef FB_ACCESS_WRAPPER
     SetupWrapProcPtr setupWrap; /* driver hook to set pixmap access wrapping */
     FinishWrapProcPtr finishWrap;       /* driver hook to clean up pixmap access wrapping */
 #endif
     DevPrivateKeyRec    gcPrivateKeyRec;
     DevPrivateKeyRec    winPrivateKeyRec;
+    unsigned char win32bpp;     /* window bpp for 32-bpp images */
+    unsigned char pix32bpp;     /* pixmap bpp for 32-bpp images */
 } FbScreenPrivRec, *FbScreenPrivPtr;
 
 #define fbGetScreenPrivate(pScreen) ((FbScreenPrivPtr) \
--- xorg-server-21.1.21/include/gcstruct.h.vanilla
+++ xorg-server-21.1.21/include/gcstruct.h
@@ -278,13 +278,8 @@
     const GCFuncs *funcs;
     GCOps *ops;
     PrivateRec *devPrivates;
-    /*
-     * The following were moved here from private storage to allow device-
-     * independent access to them from screen wrappers.
-     * --- 1997.11.03  Marc Aurele La France (tsi@xfree86.org)
-     */
-    PixmapPtr pRotatedPixmap;   /* tile/stipple rotated for alignment */
     RegionPtr pCompositeClip;
+    PixmapPtr pRotatedPixmap;   /* tile/stipple rotated for alignment */
     /* fExpose & freeCompClip defined above */
 } GC;

